src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,585 @@
+    sort.execute();
+    sorted_table = sort.get_output();
+  }
+
+  _output_segments.resize(_aggregates.size() + _groupby_column_ids.size());
+
+  /*
+   * Find all RowIDs where a value in any group by column differs from its predecessor,
+   * as those are exactly the limits of the different groups.
+   * We use a ordered set to "merge" the change locations from multiple columns.
+   * We are aware that std::set is known to be not the most efficient,
+   * but our profiling revealed that the current implementation is not a bottleneck.
+   * Currently, the vast majority of execution time is spent with sorting.
+   * Moreover, this set grows only to O(output) many elements, so inserts and lookups should stay reasonably fast.
+   *
+   *
+   * If this does become a performance problem at some point, one could try e.g. a vector of vectors,
+   * with one vector per column.
+   * Each vector stores all RowIDs where the corresponding column changes.
+   * Afterwards, the vectors have to be merged and deduplicated (presorted list merge).
+   * We have not benchmarked this solution, so we do not know how it performs in comparison to the current one.
+   *
+   * General note: aggregate_group_pointers contains the first entry of a new group,
+   *               or in other words: the last entry of a group + 1 row, similar to vector::end()
+   *
+   *               It is (in _aggregate_values) used as indicator that a group has ended just before this row.
+   *               As a result, the start of the very first group is not represented in aggregate_group_pointers,
+   *               as there was no group before that ended.
+   *               Further, the end of the last group is not represented as well.
+   *               This is because no new group starts after it.
+   *               So in total, aggregate_group_pointers will contain one element less than there are groups.
+   */
+  std::set<RowID> aggregate_group_pointers;
+  auto chunks = sorted_table->chunks();
+  for (const auto& column_id : _groupby_column_ids) {
+    auto data_type = input_table->column_data_type(column_id);
+    resolve_data_type(data_type, [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
+      std::optional<ColumnDataType> previous_value;
+      ChunkID chunk_id{0};
+      for (const auto& chunk : chunks) {
+        auto segment = chunk->get_segment(column_id);
+        segment_iterate<ColumnDataType>(*segment, [&](const auto& position) {
+          if (previous_value && position.value() != *previous_value) {
+            aggregate_group_pointers.insert(RowID{chunk_id, position.chunk_offset()});
+          }
+          // TODO(anybody) shouldnt it be sufficient to do this in an else-block?
+          // It should work both ways and as far as I know both ways are used in Hyrise
+          // With one way we can save the emplace and with the other way we save the branching.
+          previous_value.emplace(position.value());
```

Comment:
Now that I see the entire code: The else block might actually be more efficient as `emplace` always creates the object. Think of a string column, where this might be quite expensive.

Reply:
We'll probably do a small benchmark

Reply:
Update: I made a logical mistake when I thought about that. The emplace serves to update what the previous value was. This should obviously happen if we find a new value, and thus belongs into the if-block, not an else block.

Reply:
Hm, why did this not come up in any of the tests then?

Reply:
@Bensk1 tl;dr: because there was nothing wrong.
We have till now (and still do) always called emplace.
This is equivalent to calling it in both if and (the otherwise non-existing) else block.
The call in the if-block is correct.
The call in the else-block is superfluous, but since we are in the else block we know that the value has not changed, and are thus updating `previous_value` with the value it already has - so this does no harm except wasting execution time

Reply:
Just as fun fact regarding the branching:
On a presorted lineitem table (=no sorting in our operator) the group by part (=no aggregates except COUNT(*), which is fast) of TPCH-1 in `hyriseConsole` I measured the following values
```
if () {...; emplace()}                        took 750ms
if () {...} emplace()                         took 900ms
if () {...; emplace()} else {//empty}         took 1100ms
```

Reply:
> @Bensk1 tl;dr: because there was nothing wrong.

I see.

Reply:
Was this with `-O3`? Weird.